File: swap_skill_servver/src/main/java/com/example/swap_skill_servver/service/ChatService.java
package com.example.swap_skill_servver.service;

import java.util.List;
import java.util.UUID;

import com.example.swap_skill_servver.dto.ChatRoomDto;
import com.example.swap_skill_servver.dto.MessageDto;
import com.example.swap_skill_servver.model.ChatRoom;
import com.example.swap_skill_servver.model.Message;
import com.example.swap_skill_servver.model.User;

public interface ChatService {

    ChatRoom createChatRoom(User user1, User user2);

    public List<ChatRoom> findAllChatRooms();

    List<MessageDto> getMessagesForChatRoom(UUID chatRoomId);

    Message sendMessage(UUID chatRoomId, UUID senderId, String content);

    public List<ChatRoomDto> getChatRoomsForUser(UUID userId);

}

File: lib/features/chat/repository/chat_service.dart
// services/chat_service.dart

import '../models/chat_model.dart';
import '../models/message_model.dart';

abstract class ChatService {
  Future<List<Chat>> getChatRooms(String userId);

  Future<List<MessageDto>> getMessages(String chatRoomId);

  Future<Message> sendMessage(
      String chatRoomId, String senderId, String content);
}
